the error isn't a legit error object, so checking err.fatal and err.code fails. result of console.log(err) is { [Error: Connection lost: The server closed the connection.] fatal: true, code: 'PROTOCOL_CONNECTION_LOST' } which gives a TypeError: Property 'code' of object Error: Connection lost: The server closed the connection. is not a function when trying to check value of err.code